enhancements/user-monitoring.md

```diff
@@ -0,0 +1,248 @@
+a new WIP feature [4] will ensure that reconcilation of Prometheus custom resources
+will be limited to the `openshift-user-monitoring` namespace for the user monitoring prometheus operator
+and `openshift-monitoring` namespace for the cluster monitoring prometheus operator.
+
+![](user-namespaces-custom-resources.png)
+
+User monitoring Prometheus Operator (green) then will reconcile Prometheus custom resources
+in the `openshift-user-monitoring` namespace only.
+It ignores Prometheus custom resources in any other namespace.
+The same holds true for the cluster monitoring prometheus operator (red).
+
+#### Aggregation
+
+The front facing Kubernetes service has to make it as simple as possible
+for integrators to leverage both user monitoring as well as cluster monitoring.
+
+This is achieved by using Thanos Querier which allows to aggregate
+and optionally deduplicate the cluster and user monitoring metrics backends
+under single Prometheus Query endpoint.
+
+Usage of the Thanos querier component inside cluster monitoring has many advantages:
+1. Run-time deduplication of metrics available today.
+2. Potentially gap-filling of metrics series in the Prometheus HA user and cluster monitoring setup in the future.
+3. Potentially add the possibility of adding a query cache for increased performance in the future.
+
+See [5] and [6] for more details.
+
+### Tenancy
+
+![](user-monitoring-request.png)
+
+Tenancy is achieved by leveraging the existing topology that is protecting cluster monitoring Prometheus already today.
+The kube-rbac-proxy sidecar will be deployed along with prom-label-proxy in the Thanos Querier Deployment.
+
+### Risks and Mitigations
+
+Risks are mainly in the area of security, namely in the realm of accessing metrics a user has no permissions for.
+This is achieved by reusing the existing tenancy stack as well as with the newly added features for allow-lists,
+deny-lists and reconcilation limiting settings for prometheus operator.
+
+## Design Details
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+N/A
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+There is no dev preview planned, just tech preview.
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades will be orchestrated by cluster monitoring operator
+the same way as the cluster monitoring stack
+as user monitoring leverages the same semantics in terms of deployable assets
+and custom resources.
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+N/A
+
+## Alternatives
+
+N/A
```

Comment:
Sticking to current OLM based prom operator that our customers are not happy with? (I don't think that's actually an option as we're literally doing this because no one including our own SRE team is happy with the current solution, but maybe it's worth nothing that here).

Reply:
indeed, added with a declared tendency that this is not a proper replacement, plus we would need to keep OperatorGroups.